packages/ckeditor5-image/tests/imageutils.js

```diff
@@ -476,264 +476,309 @@ describe( 'ImageUtils plugin', () => {
 
-		it( 'should use the inline image type when the image.insert.type="block" option is set ' +
-			'but ImageBlockEditing plugin is not enabled', async () => {
-			const consoleWarnStub = sinon.stub( console, 'warn' );
-			const newEditor = await VirtualTestEditor.create( {
-				plugins: [ ImageUtils, ImageInlineEditing, Paragraph ],
+		describe( 'config.image.insert.type set to "block"', () => {
+			beforeEach( () => createEditor( {
+				plugins: [ ImageBlockEditing, ImageInlineEditing, Paragraph ],
 				image: { insert: { type: 'block' } }
-			} );
+			} ) );
 
-			setModelData( newEditor.model, '<paragraph>f[o]o</paragraph>' );
+			it( 'should use the block image type', () => {
+				setModelData( model, '<paragraph>f[o]o</paragraph>' );
 
-			newEditor.plugins.get( 'ImageUtils' ).insertImage();
+				editor.plugins.get( 'ImageUtils' ).insertImage( editor );
 
-			expect( consoleWarnStub.calledOnce ).to.equal( true );
-			expect( consoleWarnStub.firstCall.args[ 0 ] ).to.equal( 'image-block-plugin-required' );
-			expect( getModelData( newEditor.model ) ).to.equal( '<paragraph>f[<imageInline></imageInline>]o</paragraph>' );
+				expect( getModelData( model ) ).to.equal( '[<imageBlock></imageBlock>]<paragraph>foo</paragraph>' );
+			} );
 
-			await newEditor.destroy();
-			console.warn.restore();
-		} );
+			it( 'should use the inline image type when ImageBlockEditing plugin is not enabled', async () => {
+				const consoleWarnStub = sinon.stub( console, 'warn' );
+
+				await editor.destroy();
+				await createEditor( {
+					plugins: [ ImageInlineEditing, Paragraph ],
+					image: { insert: { type: 'block' } }
+				} );
+
+				setModelData( model, '<paragraph>f[o]o</paragraph>' );
 
-		it( 'should pass the allowed custom attributes to the inserted block image', () => {
-			setModelData( model, '[]' );
-			model.schema.extend( 'imageBlock', { allowAttributes: 'customAttribute' } );
+				editor.plugins.get( 'ImageUtils' ).insertImage();
 
-			imageUtils.insertImage( { src: 'foo', customAttribute: 'value' } );
+				expect( consoleWarnStub.calledOnce ).to.be.true;
+				expect( consoleWarnStub.firstCall.args[ 0 ] ).to.equal( 'image-block-plugin-required' );
+				expect( getModelData( model ) ).to.equal( '<paragraph>f[<imageInline></imageInline>]o</paragraph>' );
 
-			expect( getModelData( model ) )
-				.to.equal( '[<imageBlock customAttribute="value" src="foo"></imageBlock>]' );
+				console.warn.restore();
+			} );
 		} );
 
-		it( 'should omit the disallowed attributes while inserting a block image', () => {
-			setModelData( model, '[]' );
+		describe( 'config.image.insert.type set to "inline"', () => {
+			beforeEach( () => createEditor( {
+				plugins: [ ImageBlockEditing, ImageInlineEditing, Paragraph ],
+				image: { insert: { type: 'inline' } }
+			} ) );
 
-			imageUtils.insertImage( { src: 'foo', customAttribute: 'value' } );
+			it( 'should use the inline image type', () => {
+				setModelData( model, '<paragraph>f[o]o</paragraph>' );
```

Comment:
In this particular test, I think it would be more beneficial to check inserting an image into an empty paragraph. So to ensure that in the block-like context, the inserted image will be still inline. Or even better - just copy this test and change only the initial data to an empty paragraph.